Database/MySQL.php

```diff
@@ -57,11 +57,27 @@ public function __construct($params, $basePath)
 
         /* if user is set, we add authentification */
         if ($params['db_user']) {
-            $this->auth = sprintf('-u%s', $params['db_user']);
+            $cnfFile            = "[client]\n";
+            $cnfPath            = $basePath."mysql.cnf";
+            $cnfParams['user']  = $params['db_user'];
 
             if ($params['db_password']) {
-                $this->auth = sprintf("--host=\"%s\" --port=\"%d\" --user=\"%s\" --password=\"%s\"", $params['db_host'], $params['db_port'], $params['db_user'], $params['db_password']);
+                $cnfParams = array_merge(
+                    $cnfParams,
+                    array(
+                        "password" => $params['db_password'],
+                        "host" => $params['db_host'],
+                        "port" => $params['db_port']
+                    )
+                );
```

Comment:
I can rewrite it like 
`$cnfParams += array('key' => 'value')`
but this feature was provided in php 5.4. I should change composer.json to requre new version of php (old configuration is "php": ">=5.3.2") it may cause some problems with BC. If it is not a problem, i'll do this.

PS. i doesn't want to add some params like this:
`$cnfParams[key] = "value";
$cnfParams[another_key] = "another_value"
// etc`